src/AutoRest.CSharp/Common/Output/Models/Serialization/Json/JsonPropertySerialization.cs

```diff
@@ -1,40 +1,35 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
 using AutoRest.CSharp.Generation.Types;
-using AutoRest.CSharp.Output.Models.Types;
 
 namespace AutoRest.CSharp.Output.Models.Serialization.Json
 {
     internal class JsonPropertySerialization : PropertySerialization
     {
-        public JsonPropertySerialization(string parameterName, string propertyName, string serializedName, CSharpType propertyType, CSharpType? valueType, JsonSerialization valueSerialization, bool isRequired, bool shouldSkipSerialization, bool shouldSkipDeserialization, bool optionalViaNullability, string? serializationValueHook = null, string? deserializationValueHook = null)
-            : base(propertyName, serializedName, propertyType, valueType, isRequired, shouldSkipSerialization, shouldSkipDeserialization)
+        public JsonPropertySerialization(string parameterName, ValueExpression value, string serializedName, CSharpType valueType, CSharpType? serializedType, JsonSerialization valueSerialization, bool isRequired, bool shouldSkipSerialization, bool shouldSkipDeserialization, string? customSerializationMethodName = null, string? customDeserializationMethodName = null)
```

Comment:
Good point. This is a problem of the `ObjectTypeProperty.ValueType` property name, because it is not a type of the property, but a type we use in serialization/ungrouping, but I don't want to rename it cause I don't know how it is used in MPG. 

I have updated the `PropertySerialization` to use `TypedValueExpression` instead of `ValueExpression` and removed separate `ValueType` property.